zephyr/src/supp_api.c

```diff
@@ -53,7 +54,17 @@ static int send_wpa_supplicant_dummy_event(void)
 
 static inline struct wpa_supplicant * get_wpa_s_handle(const struct device *dev)
 {
-	struct wpa_supplicant *wpa_s = wpa_supplicant_get_iface(global, dev->name);
+	struct wpa_supplicant *wpa_s = NULL;
+	int ret = k_sem_take(&wpa_supplicant_ready_sem, K_SECONDS(2));
```

Comment:
Time value is quite arbitrary here. Can supplicant init fail? If not why not put `K_FOREVER`?

Reply:
It shouldn't, but don't want to block instead return failure, as blocking here would block shell.